### DISCUSSION TITLE
How can I correctly reference the config_context in a config template (jinja2)?

### DISCUSSION THREAD

**Discussion by whtkam** [netbox-community/netbox#15386] [Q&A]
In the j2 - config-template I tried

{{device.config_context.ntp-server }}
{{device.config_context.ntp_server }}

In the swagger-ui It looks like this

"config_context": {
    "ntp-servers": [
      "1.1.1.1",
      "2.2.2.2"
    ],

When I try to render it I got an error:

jinja2.exceptions.UndefinedError: 'dcim.models.devices.Device object' has no attribute 'config_context'

What ist the correct way to reference the config_context of a device in a config template

Thanks!

**Comment by candlerb** [comment]
Device mixes in ConfigContextModel: https://github.com/netbox-community/netbox/blob/v3.7.3/netbox/extras/models/configs.py#L160-L206

I believe you want `{{ device.get_config_context()["ntp_server"] }}`

If you're using it a lot, you may want to [assign it](https://jinja.palletsprojects.com/en/3.1.x/templates/#assignments) to a jinja2 variable to avoid repeatedly recalculating it.

```
{% set cc = device.get_config_context() %}
...
{{ cc["ntp_server"] }}
```

Above examples untested.

**Reply by whtkam** [reply]
Works perfectly, thank you for the support
